### PR TITLE
make CONNECTION_ID() support alias

### LINKED ISSUE #17949: CONNECTION_ID() do not support alias
### Which version of ShardingSphere did you use?
5.1.1

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy

### Expected behavior
select CONNECTION_ID() as cid;
get:
cid: 1

### Actual behavior
connection_id():  1

### PR DESCRIPTION
**Pull request by csonezp**
Fixes #17949 

Changes proposed in this pull request:
- make CONNECTION_ID() support alias

### CODE REVIEW

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowConnectionIdExecutorTest.java [hunk 1]
```diff
@@ -47,4 +72,28 @@ private ConnectionSession mockConnectionSession() {
         when(result.getConnectionId()).thenReturn(109);
         return result;
     }
+    
+    private SelectStatement mockSelectStatement() {
+
+        Collection<ProjectionSegment> projections = new LinkedList<>();
+        ProjectionsSegment segment = mock(ProjectionsSegment.class);
+        when(segment.getProjections()).thenReturn(projections);
+
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowConnectionIdExecutorTest.java:81]: Please remove useless blank line

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowConnectionIdExecutorTest.java [hunk 2]
```diff
@@ -47,4 +72,28 @@ private ConnectionSession mockConnectionSession() {
         when(result.getConnectionId()).thenReturn(109);
         return result;
     }
+    
+    private SelectStatement mockSelectStatement() {
+
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowConnectionIdExecutorTest.java:77]: Please remove useless blank line

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowConnectionIdExecutorTest.java [hunk 3]
```diff
@@ -47,4 +72,28 @@ private ConnectionSession mockConnectionSession() {
...
+    }
+    
+    private SelectStatement mockSelectStatementWithAlias() {
+        Collection<ProjectionSegment> projections = new LinkedList<>();
+        ExpressionProjectionSegment projectionSegment = new ExpressionProjectionSegment(0, 0, "connection_id()");
+        projectionSegment.setAlias(new AliasSegment(0, 0, new IdentifierValue("test_alias")));
+        projections.add(projectionSegment);
+        ProjectionsSegment segment = mock(ProjectionsSegment.class);
+        when(segment.getProjections()).thenReturn(projections);
+
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowConnectionIdExecutorTest.java:94]: Please remove useless blank line

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowConnectionIdExecutorTest.java [hunk 4]
```diff
@@ -47,4 +72,28 @@ private ConnectionSession mockConnectionSession() {
...
+    
+    private SelectStatement mockSelectStatement() {
+
+        Collection<ProjectionSegment> projections = new LinkedList<>();
+        ProjectionsSegment segment = mock(ProjectionsSegment.class);
+        when(segment.getProjections()).thenReturn(projections);
+
+        SelectStatement statement = mock(SelectStatement.class);
+        when(statement.getProjections()).thenReturn(segment);
+        return statement;
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowConnectionIdExecutorTest.java:84]: return value should name as `result`

File: shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowConnectionIdExecutorTest.java [hunk 5]
```diff
@@ -47,4 +72,28 @@ private ConnectionSession mockConnectionSession() {
...
+        Collection<ProjectionSegment> projections = new LinkedList<>();
+        ExpressionProjectionSegment projectionSegment = new ExpressionProjectionSegment(0, 0, "connection_id()");
+        projectionSegment.setAlias(new AliasSegment(0, 0, new IdentifierValue("test_alias")));
+        projections.add(projectionSegment);
+        ProjectionsSegment segment = mock(ProjectionsSegment.class);
+        when(segment.getProjections()).thenReturn(projections);
+
+        SelectStatement statement = mock(SelectStatement.class);
+        when(statement.getProjections()).thenReturn(segment);
+        return statement;
```
Comment (terrymanu) [inline shardingsphere-proxy/shardingsphere-proxy-backend/src/test/java/org/apache/shardingsphere/proxy/backend/text/admin/mysql/executor/ShowConnectionIdExecutorTest.java:97]: return value should name as `result`

**Review by terrymanu — CHANGES_REQUESTED** [review]

**Comment by csonezp** [comment]
@terrymanu done

**Review by terrymanu — APPROVED** [review]

**Pull request merged**